presentation/diarywrite/src/main/java/com/hilingual/presentation/diarywrite/component/TextScanButton.kt

```diff
@@ -0,0 +1,71 @@
+package com.hilingual.presentation.diarywrite.component
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.hilingual.core.common.extension.noRippleClickable
+import com.hilingual.core.designsystem.theme.HilingualTheme
+import com.hilingual.presentation.diarywrite.R
+
+@Composable
+internal fun TextScanButton(
+    onClick: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .clip(shape = RoundedCornerShape(8.dp))
+            .background(HilingualTheme.colors.white)
+            .border(
+                width = 1.dp,
+                color = HilingualTheme.colors.gray200,
+                shape = RoundedCornerShape(8.dp)
+            )
+            .noRippleClickable { onClick() }
```

Comment:
P3: 
```suggestion
            .noRippleClickable(onClick = onClick)
```
혹시 이 둘의 차이점을 아시나요?? 모르면 한번 공부해보기~~

Reply:
어제 효비 언니한테도 직접 듣긴 했지만 다른 팀원들이랑 미래의 제가 참고하기 위해 슬쩍 적어둡니닷 ㅎㅅㅎ

`.noRippleClickable{ onClick() }`
- onClick 자체가 이미 () -> Unit에 해당하는 람다임. 여기에서는 람다 안에서 또 다른 람다를 호출하는 구조가 되고, 이때 람다 블록이 하나 더 생기므로 객체 하나가 불필요하게 추가로 생성됨.
- noRippleClickable일 때 처리해야할 동작이 여러 개인 경우 중괄호를 사용해 함께 작성할 수 있음. 이런 경우에는 중괄호를 사용함.

`.noRippleClickable(onClick = onClick)`
- onClick에 하나의 function을 넘겨줄 때 사용함.
- 위(중괄호 사용한 방법)에서는 불필요하게 객체를 생성했으나 여기에서는 그러지 않아 미미한 차이더라도 성능 측면에서는 유리함.

Reply:
친절한 지영 언니 넘 멋있어잉..🥹
저도 사용하면서 왜 아래 방법이 권장되는 건지 잘 이해 못하고 썼던 것 같은데 덕분에 제대로 알아갑니다!!!